packages/drivers/odsp-driver/src/odspDelayLoadedDeltaStream.ts

```diff
@@ -401,6 +433,26 @@ export class OdspDelayLoadedDeltaStream {
 		return response.joinSessionResponse;
 	}
 
+	private emitDeltaConnectionUpdateEvent(metadata: Record<string, unknown>) {
+		if (this.currentConnection !== undefined) {
+			const label = metadata.sensitivityLabelsInfo as {
+				labels: unknown;
+				timestamp: string;
+			};
+			const time = Date.parse(label.timestamp);
+			if (time > this.labelUpdateTimestamp) {
+				this.labelUpdateTimestamp = time;
+				this.currentConnection.metadata = {
+					sensitivityLabelsInfo: metadata.sensitivityLabelsInfo,
+				};
+				this.currentConnection.emit(
```

Comment:
Instead of using current connection, consider putting this event on a service object (or some other long-term object).
This will address the issue of raising events on object that nobody has a hold on. It also more correct, as we might not have connection, but receive data from joinSession()

Reply:
Emitting on service as that makes the most sense if not on delta connection.